packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.connectionless-credentials.test.ts

```diff
@@ -0,0 +1,204 @@
+    await faberAgent.wallet.delete()
+    await aliceAgent.shutdown()
+    await aliceAgent.wallet.delete()
+  })
+
+  test('Faber starts with V2 W3C connection-less credential offer to Alice', async () => {
+    testLogger.test('Faber sends credential offer to Alice')
+
+    const offerOptions: OfferCredentialOptions = {
+      comment: 'V2 Out of Band offer (W3C)',
+      credentialFormats: {
+        jsonld: signCredentialOptions,
+      },
+      protocolVersion: CredentialProtocolVersion.V2,
+      connectionId: '',
+    }
+    // eslint-disable-next-line prefer-const
+    let { message, credentialRecord: faberCredentialRecord } = await faberAgent.credentials.createOutOfBandOffer(
+      offerOptions
+    )
+
+    await aliceAgent.receiveMessage(message.toJSON())
+
+    let aliceCredentialRecord = await waitForCredentialRecordSubject(aliceReplay, {
+      threadId: faberCredentialRecord.threadId,
+      state: CredentialState.OfferReceived,
+    })
+
+    testLogger.test('Alice sends credential request to Faber')
+    const acceptOfferOptions: AcceptOfferOptions = {
+      credentialRecordId: aliceCredentialRecord.id,
+    }
+    aliceCredentialRecord.protocolVersion = CredentialProtocolVersion.V2
+
+    const credentialRecord = await aliceAgent.credentials.acceptOffer(acceptOfferOptions)
+
+    testLogger.test('Faber waits for credential request from Alice')
+    faberCredentialRecord = await waitForCredentialRecordSubject(faberReplay, {
+      threadId: credentialRecord.threadId,
+      state: CredentialState.RequestReceived,
+    })
+
+    testLogger.test('Faber sends credential to Alice')
+    const options: AcceptRequestOptions = {
+      credentialRecordId: faberCredentialRecord.id,
+      comment: 'V2 Indy Credential',
+    }
+    faberCredentialRecord = await faberAgent.credentials.acceptRequest(options)
+
+    testLogger.test('Alice waits for credential from Faber')
+    aliceCredentialRecord = await waitForCredentialRecordSubject(aliceReplay, {
+      threadId: faberCredentialRecord.threadId,
+      state: CredentialState.CredentialReceived,
+    })
+
+    testLogger.test('Alice sends credential ack to Faber')
+    aliceCredentialRecord = await aliceAgent.credentials.acceptCredential(aliceCredentialRecord.id)
+
+    testLogger.test('Faber waits for credential ack from Alice')
+    faberCredentialRecord = await waitForCredentialRecordSubject(faberReplay, {
+      threadId: faberCredentialRecord.threadId,
+      state: CredentialState.Done,
+    })
+
+    expect(aliceCredentialRecord).toMatchObject({
+      type: CredentialExchangeRecord.type,
+      id: expect.any(String),
+      createdAt: expect.any(Date),
+      state: CredentialState.Done,
+      threadId: expect.any(String),
+    })
+
+    expect(faberCredentialRecord).toMatchObject({
+      type: CredentialExchangeRecord.type,
+      id: expect.any(String),
+      createdAt: expect.any(Date),
+      state: CredentialState.Done,
+      threadId: expect.any(String),
+    })
```

Comment:
Can we add some more property check here? E.g. `connectionId` is undefined, `threadId` is the same for both record, `credentials` array contains the credential record binding on the holder side, the protocol version is set to the correct version, etc...